### PR TITLE
docs: improve updating pre-commit documentation

### PR DESCRIPTION
**Pull request by 12rambau**
- there was a inconsistence between the inline and rej default. as it's inline I changed the value in the corrected section
- as users won't need the 2 options at once, I split the 2 hooks.

### CODE REVIEW

File: docs/updating.md [hunk 1]
```diff
@@ -61,7 +61,18 @@ setting you use.
 
 ## Preventing Commit of Merge Conflicts
 
-If you use `--conflict rej` (the default):
+If you use `--conflict inline` (the default) then you need to check for conflicts marks in your files:
```
Comment (sisp) [inline docs/updating.md:64]: * `marks` ->  `markers`
* An empty line is missing.
* The line length exceeds 88 characters, so [Prettier won't be happy](https://github.com/copier-org/copier/blob/5fa3b9fb772101ac67476ab235b8e66ca26900de/.prettierrc.yml#L1).

```suggestion
If you use `--conflict inline` (the default) then you need to check for conflicts
markers in your files:

```

File: docs/updating.md [hunk 2]
```diff
@@ -61,7 +61,18 @@ setting you use.
...
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v4.3.0
+      hooks:
+          # Prevent committing inline conflict markers
+          - id: check-merge-conflict
+            args: [--assume-in-merge]
+```
+
+If you use `--conflict rej` Then you need to reject all the generated `.rej` files:
```
Comment (sisp) [inline docs/updating.md:75]: Some spelling, wording and content improvements:

```suggestion
If you use `--conflict rej` then you need to review and remove all generated `.rej`
files:
```

**Review by sisp — CHANGES_REQUESTED** [review]
Thanks for your contribution and also for catching the outdated default value for the `--conflict` flag, @12rambau! :pray:

I have only a few minor remarks.

**Review by yajo — CHANGES_REQUESTED** [review]
I think it's good to keep both examples together. In a shared project, one user might use  rej mode, while others use inline mode. In any case, you don't want those files committed.

If you want to keep them split because of clarity, at least please explain this point too.

**Comment by bswck** [comment]
The current documentation is confusing. At first it is stated that `--conflict inline` is the default, and later that `--conflict rej` is.

**Comment by yajo** [comment]
In that case, the default is inline.

**Review by yajo — APPROVED** [review]

File: docs/updating.md [hunk 3]
```diff
@@ -70,17 +84,17 @@ repos:
           # Prevent committing .rej files
           - id: forbidden-files
             name: forbidden files
-            entry: found Copier update rejection files; review them and remove them
+            entry: found Copier update rejection files; review and remove them before merging.
```
Comment (sisp) [inline docs/updating.md:87]: This was actually incorrect before: [`entry`](https://pre-commit.com/#hooks-entry) refers to the executable to run. This is a [description](https://pre-commit.com/#hooks-description):

```suggestion
            description: found Copier update rejection files; review and remove them before merging.
```

File: docs/updating.md [hunk 4]
```diff
@@ -70,17 +84,17 @@ repos:
...
-          # Prevent committing inline conflict markers
-          - id: check-merge-conflict
-            args: [--assume-in-merge]
 ```
 
+!!! note
+
+    For shared project that use both "rej" and "inline" depending on the user,
+    you can add both repo to you `pre-commit-config.yaml` file, making sure that none
+    of these files are committed.
```
Comment (sisp) [inline docs/updating.md:96]: Just some minor suggestions for better clarity and grammatical correctness:

```suggestion
    For projects that use both `rej` and `inline` depending on each user's preference,
    you can add both hooks to your `pre-commit-config.yaml` file, making sure that no
    unresolved merge conflicts are committed.
```

**Review by sisp — CHANGES_REQUESTED** [review]
Just some final polishing. :wink:

**Comment by sisp** [comment]
Oh, and CI is failing because Prettier wants a line wrap in the `pre-commit-config.yaml` snippet. It's probably best to make this a proper multiline YAML string:

```yaml
description: >
  found Copier ...
  ...
```

**Comment by yajo** [comment]
Fixed

**Review by sisp — APPROVED** [review]
Perfect, thanks! :ok_hand:

**Pull request merged**